AK/Statistics.h

```diff
@@ -27,10 +31,23 @@ class Statistics {
     }
 
     T const sum() const { return m_sum; }
-    float average() const { return (float)sum() / size(); }
+
+    // FIXME: Unclear Wording, average can mean a lot of different things
+    // Median, Arithmetic Mean (which this is), Geometric Mean, Harmonic Mean etc
+    float average() const
+    {
+        // Let's assume the average of an empty dataset is 0
+        if (size() == 0)
+            return 0;
+
+        return (float)sum() / size();
```

Comment:
note: not sure about use cases etc., but shall we deal with `sum` going out of bounds in any way? (integer overflow?)

Reply:
There's a TODO about this already iirc?

Reply:
I checked there is one on line 28 and it was there already before this PR.

I thought about how i could fix it when i originally started on this, didn't find a good solution because of the genericness of the class. (How would you even check for overflow if you're dealing with different sizes of ints and floats based on templates.).
So if you have a good idea on how i could deal with it i'd add it to the PR but if not i think we should just leave it open for the future.

Reply:
if you want, you might place a comment there (a `TODO`?) to signal, that weʼre aware of potential overflow problems while computing the sum. yet itʼs not, by all means, necessary!

Reply:
> if you want, you might place a comment there (a `TODO`?) to signal, that weʼre aware of potential overflow problems while computing the sum/average. yet itʼs not, by all means, necessary!

I think having two todos for exactly the same thing inside the same file is redundant and confusing. When it gets resolved someone might forget to remove the other todos etc.

Reply:
not exactly the same for me… but by all means I wonʼt insist on that! 😅

(the thing is that `sum` here is some kind of a shortcut here; you might want to sum partially and reduce with division etc.; sum itself needs just data store big enough to hold the result — those are, of course, speculations so let us ignore them for now.)

resolving! 😃

Reply:
Okay, I'll add notes there when I have time.

Reply:
done